### PR TITLE
Add role and purpose for subnet settings

### PR DESCRIPTION
**Pull request by jawnsy**
Closes: #339

### CODE REVIEW

File: modules/subnets/main.tf [hunk 1]
```diff
@@ -31,6 +31,8 @@ resource "google_compute_subnetwork" "subnetwork" {
   ip_cidr_range            = each.value.subnet_ip
   region                   = each.value.subnet_region
   private_ip_google_access = lookup(each.value, "subnet_private_access", "false")
+  purpose                  = lookup(each.value, "subnet_purpose", null)
```
Comment (jawnsy) [inline modules/subnets/main.tf:34]: followed the `subnet_xxx` pattern here, but `subnet-beta` declares these as just `role` and `purpose` (no prefix)

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @jawnsy 
We will also need to update versions in [main module](https://github.com/terraform-google-modules/terraform-google-network/blob/master/versions.tf) and [subnets](https://github.com/terraform-google-modules/terraform-google-network/blob/a7b91bc1a70c323b99eecc229a70b449b6bda401/modules/subnets/versions.tf#L23) to minimum 3.83

File: modules/subnets/main.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/subnets/main.tf:34]: We can go with subnet prefix since others follow this pattern in this module

**Comment by jawnsy** [comment]
@bharathkkb Thanks for the quick review and the pointers to the code that needed to be updated! I've bumped the minimum versions to 3.83

**Pull request closed without merging**